test/functional/e2e_test.go

```diff
@@ -40,7 +45,25 @@ func init() {
 	}
 }
 
-func runTest(t *testing.T, name string, overridenCriHandlers map[string]interface{}, testFunction func(*testing.T, api.RuntimeServiceClient, context.Context)) {
+type testEnv struct {
+	t           *testing.T
+	handlers    map[string]interface{}
+	client      api.RuntimeServiceClient
+	forceConfig string
+	mgr         resmgr.ResourceManager
+	cache       cache.Cache
+}
+
+// a big freakin' test lock to protect until we fix racy components...
+var BFTL sync.Mutex
```

Comment:
Is this big hammer still needed? If it is, what are the racy components left? We should create an issue tracking those.

Reply:
And if we can expose those bugs by running this test without the lock, even better!

Reply:
> Is this big hammer still needed? 

Hopefully not. Let's remove the BFTL and find out...